lib/rouge/lexers/opentypefeature.rb

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# http://www.adobe.com/devnet/opentype/afdko/topic_feature_file_syntax.html
+
+module Rouge
+  module Lexers
+    class OpenTypeFeature < RegexLexer
+      title "OpenTypeFeature"
+      desc "OpenType feature code"
+      tag 'opentypefeature'
+      aliases 'feature', 'fea', 'opentype'
+      filenames '*.fea'
+
+      def self.keywords
+        @keywords ||= %w(
+          Ascender Attach CapHeight CaretOffset CodePageRange Descender FontRevision
+          GlyphClassDef HorizAxis.BaseScriptList HorizAxis.BaseTagList HorizAxis.MinMax
+          IgnoreBaseGlyphs IgnoreLigatures IgnoreMarks LigatureCaretByDev LigatureCaretByIndex
+          LigatureCaretByPos LineGap MarkAttachClass MarkAttachmentType NULL Panose RightToLeft
+          TypoAscender TypoDescender TypoLineGap UnicodeRange UseMarkFilteringSet Vendor
+          VertAdvanceY VertAxis.BaseScriptList VertAxis.BaseTagList VertAxis.MinMax VertOriginY
+          VertTypoAscender VertTypoDescender VertTypoLineGap XHeight
+
+          anchorDef anchor anonymous anon by contour cursive device enumerate enum
+          exclude_dflt featureNames feature from ignore include_dflt include languagesystem
+          language lookupflag lookup markClass mark nameid name parameters position pos required
+          reversesub rsub script sizemenuname substitute subtable sub table useExtension
+          valueRecordDef winAscent winDescent
+        )
+      end
+
+
+      identifier = /[a-z_][a-z0-9\/_]*/i
+
+      state :root do
+        rule /\n+/m, Text
+        rule /\s+/, Text::Whitespace
+        rule /#.*$/, Comment
+
+        # feature <tag>
+        rule /(anonymous|anon|feature|lookup|table)((?:\s|\\\s)+)/ do
+          groups Keyword, Text
+          push :featurename
+        end
+        # } <tag> ;
+        rule /(\})((?:\s|\\\s)*)((?:\s|\\\s)*)/ do
+          groups Punctuation, Text
+          push :featurename
+        end
+        # solve include( ../path)
+        rule /(include)/i, Keyword, :includepath
+
+        rule /[\[\]\/(\){},.:;-=%*<>']/, Punctuation
```

Comment:
You don't need to escape the `)` when it's in a character class. On the other hand, you should move `-` to the beginning of the character class as it has a different meaning when occurring partway through.

Reply:
It's not harmful but just so you know, you don't need to escape `-` if it's the first character in a character class but you do need to if it is not the first character :)